src/overlay/Peer.h

```diff
@@ -53,8 +53,32 @@ class Peer : public std::enable_shared_from_this<Peer>,
 {
 
   public:
+    static constexpr uint32_t FIRST_VERSION_SUPPORTING_FLOW_CONTROL = 20;
+
     typedef std::shared_ptr<Peer> pointer;
 
+    // Data structure that records the time it takes to process a message and
+    // sends the signal back to the peer. Callers must manually call
+    // `finishProcessing`.
+    struct MessageTracker
```

Comment:
`MessageTracker` is a bit misleading.

What this thing is doing really is:
* refcount usage
* invoke a callback with some context when that refcount gets to zero

It may be useful to have this tracker "own" some other data to add to the context for the callback etc, like a `StellarMessage` or a `shared_ptr<StellarMessage>` (depending on context), then it allows you to replace function calls like
foo(StellarMessage const& msg, MessageTracker cb) by `foo(TrackerPtr<StellarMessage> msgTracker)`, and access the store object with `msgTracker.getData()`